src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -144,29 +145,20 @@ public static void placeConstructionTape(final BlockPos pos, final Tuple<Tuple<I
 
     public static int checkIfPlaceable(@NotNull final int x, @NotNull final int y, @NotNull final int z, @NotNull final World world)
     {
-        int newY = y;
-        boolean working = true;
-        while (working)
+        BlockPos target = new BlockPos(x,y,z);
+        final Chunk chunk = world.getChunkFromBlockCoords(target);
+
+        target = new BlockPos(x, chunk.getTopFilledSegment() + 16, z);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)